app/services/setting_registry.rb

```diff
@@ -0,0 +1,96 @@
+class SettingRegistry
+  include Singleton
+
+  def initialize
+    @settings = {}
+    load
+  end
+
+  def logger
+    Foreman::Logging.logger('app')
+  end
+
+  def find(name)
+    @settings[name.to_s]
+  end
+
+  # Returns a setting effective value
+  # Call as `Foreman.settings[<name>]`
+  #
+  #   Foreman.settings['default_locale'] => nil.
+  #   Foreman.settings[:default_locale] => nil.
+  #
+  def [](name)
+    find(name)&.value
+  end
+
+  def update(name, value)
```

Comment:
Yeah, I guess we can, I just didn't want to make it a public API and `[]=` looks much more as a public API :) but why not ;)

Reply:
why not make it public? long term I see the registry as the main point of interaction with settings, so that if you want to set a value the best practice would be using this method